src/btree/bt_sync.c

```diff
@@ -148,16 +149,27 @@ __sync_delete_obsolete_ref(WT_SESSION_IMPL *session, WT_REF *ref)
         return (0);
```

Comment:
If the btree is clean, do we run checkpoint cleanup on it?

Reply:
Currently, if the entire tree is removed using a timestamp, we allow checkpoint cleanup to process the tree even if it is clean. I believe this logic can be enhanced. Still, it doesn't work for non-timestamped tables.